File: app/src/main/java/com/hqq/example/ui/jetpack/hilt/AnalyticsServiceImpl.java
package com.hqq.example.ui.jetpack.hilt;

import javax.inject.Inject;

/**
 * @Author : huangqiqiang
 * @Package : com.hqq.example.ui.jetpack.hilt
 * @FileName :   AnalyticsServiceImpl
 * @Date : 2020/8/13 0013  下午 4:19
 * @Email : qiqiang213@gmail.com
 * @Descrive :
 */
public class AnalyticsServiceImpl implements AnalyticsService {

    @Inject
    AnalyticsServiceImpl() {
    }

    @Override
    public String analyticsMethods() {
        return "AnalyticsServiceImpl";
    }


}

File: app/src/main/java/com/hqq/example/ui/jetpack/di/hilt/AnalyticsServiceImpl.kt
package com.hqq.example.ui.jetpack.di.hilt

import javax.inject.Inject

/**
 * @Author : huangqiqiang
 * @Package : com.hqq.example.ui.jetpack.di.hilt
 * @FileName :   AnalyticsServiceImpl
 * @Date : 2020/8/13 0013  下午 4:19
 * @Email : qiqiang213@gmail.com
 * @Descrive :
 */
class AnalyticsServiceImpl @Inject internal constructor() : AnalyticsService {
    override fun analyticsMethods(): String {
        return " 我实现的是AnalyticsServiceImpl"
    }
}